### DISCUSSION TITLE
Simple feature suggestion for experienced producers

### DISCUSSION THREAD

**Discussion by nezvers** [YuriSizov/boscaceoil-blue#41] [General]
I was aware of the original version but it was too clunky to do anything serious. Bosca Ceoil still has an image on internet that you can't make a serious music with it. Recently I was researching tools I could use to teach music in a game developer workshop and sadly I decided to not feature Bosca Ceoil because of lack of these features.

## Note selecting and dragging
It is important to iterate as fast as possible. Manual removal one by one then redraw one by one is not fast and pleasant.

## Note velocity
Without velocity control it can't be used as a proper songwriting tool and will sound like a toy.

## Pattern length
One bar length for a pattern is too short to have any meaningful expression. It's impossible to write down a chord progression or rushed chord accent.
I'd expect to behave same way as notes (currently with mouse wheel).

## Triplets, 1/32 notes, and 1/64 notes
Those are another essential songwriting tools.

## Reference software
Please, try out free FL Studio. Just midi note writing aspect.
I use different DAWs for music writing, recording and engineering, but when it comes to midi writing, nothing beats FL studio experience.
It has nice synergy with modifier keys (shift, ctrl, alt) for everything.
- ctrl + click = selection rectangle
- Shift + click = note with shortest pattern cell length
- Note end dragging = resize note or all selected notes by same amount
- shift + drag = clone note or selection
- right click drag over velocities = match velocities to dragged line
- alt + mouse wheel = adjust velocity for hovered note or selection
- shift + mouse wheel = offset hovered note or selection position in time

![image](https://github.com/user-attachments/assets/ca3390db-9850-4dbd-9f78-7b4704dae4ec)

**Comment by YuriSizov** [comment]
Hey, thanks for your interest. May I ask what makes you interest in Bosca in the first place? I understand what you think is lacking, but I'd like to understand what makes you consider Bosca when other options seem to cover your needs better. Could you elaborate on that?

The idea behind this app is to be simple and approachable. It certainly could get more complex, but it doesn't try to cover all the basics of music production. And yes, the range of songs you could write with it is limited, and the time scale is pretty fixed, and it all mostly sounds like synth music from old consoles. That's fine though, that's well within what makes it special to so many folks who still use it 10 years after the release.

So while I appreciate the areas where it's lacking, I'm not sure what your use case for Bosca is right now based on your needs.

**Reply by nezvers** [reply]
My use case would be using it as a teaching tool I can give to students or show in tutorial videos.
Currently for that I use Reaper, since it's cross-platform, "free" and goes beyond basic needs. But Reaper requires installing VST/ LV2 or CLAP plugins to be useful. That is a pain in the butt of itself, to go through for everyone that's jumping in for the first time.
LMMS is really close to recommend, but it's note writing is the worst I have used.
Mentioned features are the only reason I can't recommend Bosca Ceoil. When I got the news of Blue version my hopes went up that it could meet the threshold.
